### ISSUE TITLE
[DEV]nodeset <noderange> shell does not work until osimage is assigned to the node

### ISSUE THREAD

**Issue by whowutwut** [xcat2/xcat-core#886]
Tried this using Virtual Machines... I'm not sure if the behavior is identical for a bare metal node.

``nodeset <noderange> shell`` , does not seem to work until an osimage is associated with the node.  Is this working as designed, @zet809?   

The goal here was to try and get the VM booted into the genesis kernel.  Didn't expect to need to associate a osimage with the node... 

```
[root@c910f02c05p03 etc]# nodeset fs2vm102 osimage=rhels7.2-ppc64le-install-compute
fs2vm102: Error: Unable to identify plugin for this command, check relevant tables: noderes.netboot=(grub2[-]?.*);nodehm.mgt
[root@c910f02c05p03 etc]# nodeset fs2vm102 shell
fs2vm102: Error: Unable to identify plugin for this command, check relevant tables: noderes.netboot=(grub2[-]?.*);nodehm.mgt

[root@c910f02c05p03 etc]# chdef fs2vm102 netboot=grub2
1 object definitions have been created or modified.
[root@c910f02c05p03 etc]# nodeset fs2vm102 shell
fs2vm102: shell
Error: stop configuration, missing /tftpboot/boot/grub2/grub2..

[root@c910f02c05p03 etc]# nodeset fs2vm102 osimage=rhels7.2-ppc64le-install-compute
fs2vm102: install rhels7.2-ppc64le-compute
[root@c910f02c05p03 etc]# nodeset fs2vm102 shell
fs2vm102: shell
```

**Comment by zet809** [comment]
@whowutwut Have got the point, will fix it soon.

**Comment by zet809** [comment]
@whowutwut Have got the point, will fix it soon.

**Comment by gurevichmark** [comment]
When "netboot" attr is set to grub2, the grub2.pm is called. It checks the "arch" attr of **osimage** specified on **nodeset** command. This "arch" attr is used to make a link grub2-_compute node_ to grub2._arch_ in /tftpboot/boot/grub2 directory. 
However if "shell" or "shutdown" specified for **nodeset** command, there is no **osimage** and therefore no "arch" specified. The error message is displayed.

Talked to Xiao Peng about possible solution. He agreed that if no **osimage** specified for the **nodeset** command, we can use the "arch" attribute from the _compute node_ definition.

**Comment by gurevichmark** [comment]
Fixed by pull request #936

**Comment by whowutwut** [comment]
Using the following build for Verification: 
`Version 2.12 (git commit bb5988332333745cbe759f5c8299c8379170b994, built Mon May  9 09:30:39 EDT 2016)`

Create a bare VM just for this test: 
```
[root@fs2vm100 scripts]# cat fs2vm113.stanza
# <xCAT data object stanza file>

fs2vm113:
    objtype=node
    arch=ppc64le
    cons=kvm
    groups=vm,all
    ip=10.3.31.113
    mgt=kvm
    netboot=grub2
    serialport=0
    serialspeed=115200
    vmcpus=2
    vmhost=fs2
    vmmemory=8192
    vmnicnicmodel=virtio
    vmnics=br0,br1,br2,br3
    vmstorage=dir:///var/lib/libvirt/images

[root@fs2vm100 scripts]# cat fs2vm113.stanza | mkdef -z
1 object definitions have been created or modified.
```

Then create the VMs
```
[root@fs2vm100 scripts]# makehosts fs2vm113
[root@fs2vm100 scripts]# mkvm fs2vm113 -s 15G
[root@fs2vm100 scripts]# makedhcp fs2vm113
```

Running the nodeset shell: 

```
[root@fs2vm100 scripts]# nodeset fs2vm112 shell
fs2vm112: shell
```

Looks good.  Thanks
